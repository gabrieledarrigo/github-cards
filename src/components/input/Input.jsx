import React from 'react';
import PropTypes from 'prop-types';
import styles from './Input.module.css';

export const InputType = {
  TEXT: 'text',
  SEARCH: 'search',
};

export default function Input({
  name,
  type,
  value,
  placeholder,
  required,
  onChange,
  ...rest
}) {
  return (
    <input
      className={styles.input}
      name={name}
      type={type}
      placeholder={placeholder}
      value={value}
      required={required}
      onChange={onChange}
      {...rest}
    />
  );
}

Input.propTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.oneOf([
    InputType.TEXT,
  ]).isRequired,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
  onChange: PropTypes.func,
};

Input.defaultProps = {
  placeholder: null,
  required: false,
  onChange: () => {},
};
