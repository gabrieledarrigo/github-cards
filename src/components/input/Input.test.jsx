import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Input, { InputType } from './Input';

describe('Input', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <Input
        name="input-name"
        type={InputType.TEXT}
        value="A value"
        placeholder="This is the placeholder"
        onChange={() => {}}
        required
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should invoke onChange function when a user change input value', () => {
    const wrapper = shallow(
      <Input
        name="input-name"
        type={InputType.TEXT}
        value="A value"
        placeholder="This is the placeholder"
        onChange={() => {}}
        required
      />,
    );

    wrapper.simulate('change', {
      e: {
        target: {
          value: 'Changed value',
        },
      },
    });
  });
});
