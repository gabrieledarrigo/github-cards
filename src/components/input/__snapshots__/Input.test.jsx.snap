// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<input
  className="input"
  name="input-name"
  onChange={[Function]}
  placeholder="This is the placeholder"
  required={true}
  type="text"
  value="A value"
/>
`;
